<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="description" content="CSC2417 - Algorithms &amp; Genome Analysis">

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>CSC2417</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/jts/csc2417">View on GitHub</a>

          <h1 id="project_title">CSC2417</h1>
          <h2 id="project_tagline">CSC2417 - Algorithms &amp; Genome Analysis</h2>

        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h3>
            <a id="csc2417---algorithms--genome-analysis" class="anchor" href="#csc2417---algorithms--genome-analysis" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Course Description</h3>

<b>Instructor:</b> <a href=http://labs.oicr.on.ca/simpson-lab>Jared Simpson</a>
<b>Office:</b> <a href=https://goo.gl/maps/eNWahohBCGo>OICR</a> 6-27
<b>Office hours:</b> By appointment

<p>In this course we will explore the computational problems that have emerged from genome sequencing. The topics will include the string algorithms used for the classic sequence alignment and genome assembly problems, algorithms for comparing genomes, sequence classification using probabilistic models and algorithms for analysing gene expression data. Throughout the course a special emphasis will be made on efficient algorithms designed to meet the challenges of rapidly growing data sets in the current era of high-throughput genome sequencing. The course is intended for Computer Science graduate students, and all of the required biology will be explained in the class. Students in biological and related sciences with a strong computational background are encouraged to participate.</p>

<h3>
<a id="Grading" class="anchor" href="#grading" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Grading</h3>

<p>
This course will be assessed by three written assignments (60%) and a final project (40%). The final project will in the form of a paper on a topic selected by the student relating to sequence analysis. This can be a review of recently published algorithms, an original idea or an implementation of a significant algorithm. Detailed instructions will be posted later this month.
</p>

<h3>
<a id="syllabus" class="anchor" href="#syllabus" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Syllabus</h3>

<ol>
<li>Sept 14 - Introduction to Genomes and Genome sequencing</li>
<li>Sept 21 - No lecture</li>
<li>Sept 28 - Strings - Introduction to strings and matching problems</li>
<li>Oct 5 - Sequence Alignment 1 - Introduction to alignment and dynamic programming</li>
<li>Oct 12 - Sequence Alignment 2 - Indexing data structures, the suffix array and FM-index</li>
<li>Oct 19 - Sequence Alignment 3 - Multiple alignment and Phylogenetic reconstruction</li>
<li>Oct 26 - Genome Assembly 1 - Assembly Graphs and basic algorithms</li>
<li>Nov 2 - Genome Assembly 2 - Compressed graph representations/Fast construction</li>
<li>Nov 9 - Genome Analysis 1 - Basic sequence classification</li>
<li>Nov 16 - Genome Analysis 2 - Genome Annotation</li>
<li>Nov 23 - Gene expression 1 - Alignment-based methods</li>
<li>Nov 30 - Gene expression 2 - Alignment-free methods</li>
<li>Dec 7 - Recap and discussion of current research topics</li>
</ol>

<h3>
<a id="materials" class="anchor" href="#materials" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Teaching Materials
</h3>

<p>
There is no required textbook for this course but the classic reference for much of the material is <a href=http://www.cambridge.org/ca/academic/subjects/life-sciences/genomics-bioinformatics-and-systems-biology/biological-sequence-analysis-probabilistic-models-proteins-and-nucleic-acids>Biological Sequence Analysis</a> by Durbin, Eddy, Krogh and Mitchison. The lectures are designed to stand alone but this book is well worth reading to supplement the course's content.
<br>
Many of the lectures reuse slides with permission from Ben Langmead's excellent <a href=http://www.langmead-lab.org/teaching-materials/>teaching materials</a>. In the lectures posted above Ben's slides retain the John Hopkins University logo.
</p>

<h3>
<a id="discussion group" class="anchor" href="#discussion" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Discussion Group
</h3>

<p>
There is a <a href=https://groups.google.com/forum/#!forum/csc2417-2016>google group</a> for the course where you can ask questions and receive help. Please sign up here as it will also be used for course announcements.
</p>

<h3>
<a id="office hours" class="anchor" href="#office hours" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Office Hours
</h3>

<p>
The course instructor is located at <a href=https://goo.gl/maps/eNWahohBCGo>OICR</a>. I am available to talk to students most days of the week but you should email first to make sure I am free.
</p>

      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p class="copyright">CSC2417 maintained by <a href="https://github.com/jts">jts</a></p>
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
